### PR TITLE
added test suite for c/miscMultipleTemplates

### PR DESCRIPTION
**Pull request by pchittum**
Adds test suite for c/miscMultipleTemplates

### CODE REVIEW

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            is: MiscMultipleTemplates
+        });
+        document.body.appendChild(element);
+
+        // Simulate user click
+        const clickEvent = new CustomEvent('click');
+        const button = element.shadowRoot.querySelector('lightning-button');
+        button.dispatchEvent(clickEvent);
+
+        // After click, DOM is updated showing templateTwo
```
Comment (muenzpraeger) [inline force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js:43]: Insert standard text before promises.

```
            // Return a promise to wait for any asynchronous DOM updates. Jest
            // will automatically wait for the Promise chain to complete before
            // ending the test and fail the test if the promise rejects.
```

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        const TEMPLATE_TEXT_CONTENT = 'Template One';
+
+        // Create initial element
+        const element = createElement('c-misc-multiple-templates', {
+            is: MiscMultipleTemplates
+        });
+        document.body.appendChild(element);
+
+        // On initial render templateOne should be displayed
+        const pElements = element.shadowRoot.querySelectorAll('p');
```
Comment (muenzpraeger) [inline force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js:22]: Naming for elements (for consistency):

Elements in in "El" (singular) or "Els" (plural), i. e. `pEls`. 

FYI @pozil

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        const [pTextElement] = [...pElements].filter(
+            (item) => item.textContent === 'Template One'
+        );
+        expect(pTextElement.textContent).toBe(TEMPLATE_TEXT_CONTENT);
```
Comment (muenzpraeger) [inline force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js:26]: We should also check for the image, as that's an important visual element for this component.

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        const button = element.shadowRoot.querySelector('lightning-button');
+        button.dispatchEvent(clickEvent);
+
+        // After click, DOM is updated showing templateTwo
+        return Promise.resolve().then(() => {
+            const pElements = element.shadowRoot.querySelectorAll('p');
+            const [pTextElement] = [...pElements].filter(
+                (item) => item.textContent === 'Template Two'
+            );
+            expect(pTextElement.textContent).toBe(TEMPLATE_TEXT_CONTENT);
```
Comment (muenzpraeger) [inline force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js:49]: We should also check for the image, as that's an important visual element for this component.

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+        // On initial render templateOne should be displayed
+        const pElements = element.shadowRoot.querySelectorAll('p');
+        const [pTextElement] = [...pElements].filter(
+            (item) => item.textContent === 'Template One'
+        );
+        expect(pTextElement.textContent).toBe(TEMPLATE_TEXT_CONTENT);
+    });
+
+    it('displays templateTwo on click', () => {
```
Comment (muenzpraeger) [inline force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js:29]: We'd need another test that tests for switching back to template 1.

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js (hunk 1 above)
Comment (pchittum) [reply force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js]: done.

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js (hunk 2 above)
Comment (pchittum) [reply force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js]: done

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js (hunk 4 above)
Comment (pchittum) [reply force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js]: done

File: force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js (hunk 5 above)
Comment (pchittum) [reply force-app/main/default/lwc/miscMultipleTemplates/__tests__/miscMultipleTemplates.test.js:39]: done.

**Pull request merged**